src/files-breadcrumbs.tsx

```diff
@@ -51,6 +58,126 @@ function useHostname() {
     return hostname;
 }
 
+function BookmarkButton({ path }: { path: string[] }) {
+    const [isOpen, setIsOpen] = React.useState(false);
+    const [user, setUser] = React.useState<cockpit.UserInfo | null>(null);
+    const [bookmarks, setBookmarks] = React.useState<string[]>([]);
+    const [bookmarkHandle, setBookmarkHandle] = React.useState<cockpit.FileHandle<string> | null>(null);
+
+    const currentPath = "file://" + (path.join("/") || "/");
+    const defaultBookmarks = [];
+    if (user?.home)
+        defaultBookmarks.push({ name: _("Home"), loc: user?.home }); // TODO: add trash
+
+    useInit(async () => {
+        const user_info = await cockpit.user();
+        setUser(user_info);
+
+        const handle = cockpit.file(`${user_info.home}/.config/gtk-3.0/bookmarks`);
+        setBookmarkHandle(handle);
+
+        handle.watch((content) => {
+            if (content !== null) {
+                setBookmarks(content.split("\n").filter(line => line.startsWith("file://")));
+            } else {
+                setBookmarks([]);
+            }
+        });
+
+        return [handle];
+    });
+
+    const saveBookmark = async () => {
+        cockpit.assert(user !== null, "user is null while saving bookmarks");
+        cockpit.assert(bookmarkHandle !== null, "bookmarkHandle is null while saving bookmarks");
+        const bookmark_file = basename(bookmarkHandle.path);
+        const config_dir = bookmarkHandle.path.replace(bookmark_file, "");
+
+        try {
+            await cockpit.spawn(["mkdir", "-p", config_dir]);
+        } catch (err) {
+            console.error("unable to create bookmark config directory", err);
+            return;
+        }
+
+        try {
+            await bookmarkHandle.modify((old_content: string) => {
+                if (bookmarks.includes(currentPath)) {
+                    return old_content.replace(`${currentPath}\n`, "");
+                } else {
+                    if (old_content === null) {
+                        return `${currentPath}\n`;
+                    } else {
+                        return old_content + `${currentPath}\n`;
```

Comment:
This added line is not executed by any test.

Reply:
This should be tested, is this a coverage issue or a missing test?